src/site/_includes/partials/head.njk

```diff
@@ -32,15 +32,34 @@
 <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16.png">
 <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon.png">
 <link rel="mask-icon" color="#0054ff" href="/images/safari-pinned-tab.svg">
-<script type="module" src="{{ helpers.hashForProd('/js/app.js') }}"></script>
+{% set scriptLoader %}
+function loadScript(url, type) {
+  const s = document.createElement('script');
+  s.src = url;
+  if (type) s.type = type;
+  if (type === 'module') {
+    const pre = document.createElement('link');
+    pre.rel = 'modulepreload';
+    pre.href = url;
+    document.head.appendChild(pre);
+    window.addEventListener('DOMContentLoaded', () => {
+      document.head.appendChild(s);
+    })
```

Comment:
I was experiencing errors on `/live/`, my hypothesis was that modules are not deferred when they are dynamically loaded like this.

Changing this code to load the modules after `DOMContentLoaded` appears to have fixed the issues.

Reply:
That seems right. I'd probably add a comment explaining it since the logic might be confusing to a future reader.

```
{# We use DOMContentLoaded as the loader script is running sync, and inserting a module script here doesn't defer. This brings back normal type="module" behavior. #}
```